components/inputs/input-number.js

```diff
@@ -99,42 +92,84 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 		this.maxExclusive = false;
 		this.minExclusive = false;
 		this.required = false;
-		this.trailingZeroes = false;
 
 		this._formattedValue = '';
 		this._inputId = getUniqueId();
-		this._numDecimalDigits = 0;
+		this._trailingZeroes = false;
+		this._valueTrailingZeroes = '';
+	}
+
+	get maxFractionDigits() {
+		return this._maxFractionDigits ?? Math.max(this.minFractionDigits, 3);
+	}
+	set maxFractionDigits(val) {
+		if (isNaN(val) || val < 0 || val > 20 || val < this.minFractionDigits) {
+			throw new RangeError('maxFractionDigits must be between 0 and 20 and >= minFractionDigits');
+		}
```

Comment:
Also added some exceptions here for invalid min/max fraction digits. I don't believe this is going to break anything that wasn't already broken, but this allows us to avoid some `try`-`catch` scenarios elsewhere and also helps consumers realize that they're doing something weird. Previously it would have just not done anything.